interop-testing/src/main/java/io/grpc/testing/integration/GrpclbFallbackTestClient.java

```diff
@@ -0,0 +1,302 @@
+      } else {
+        System.err.println("Unknown argument: " + key);
+        usage = true;
+        break;
+      }
+    }
+    if (usage) {
+      GrpclbFallbackTestClient c = new GrpclbFallbackTestClient();
+      System.out.println(
+          "Usage: [ARGS...]"
+          + "\n"
+          + "\n  --server_uri                          Server target. Default: "
+          + c.serverUri
+          + "\n  --custom_credentials_type             Name of Credentials to use. "
+          + "Default: " + c.customCredentialsType
+          + "\n  --unroute_lb_and_backend_addrs_cmd    Shell command used to make "
+          + "LB and backend addresses unroutable. Default: "
+          + c.unrouteLbAndBackendAddrsCmd
+          + "\n  --blackhole_lb_and_backend_addrs_cmd  Shell command used to make "
+          + "LB and backend addresses black holed. Default: "
+          + c.blackholeLbAndBackendAddrsCmd
+          + "\n  --test_case=TEST_CASE        Test case to run. Valid options are:"
+          + "\n      fast_fallback_before_startup : fallback before LB connection"
+          + "\n      fast_fallback_after_startup : fallback after startup due to "
+          + "LB/backend addresses becoming unroutable"
+          + "\n      slow_fallback_before_startup : fallback before LB connection "
+          + "due to LB/backend addresses being blackholed"
+          + "\n      slow_fallback_after_startup : fallback after startup due to "
+          + "LB/backend addresses becoming blackholed"
+          + "\n      Default: " + c.testCase
+      );
+      System.exit(1);
+    }
+  }
+
+  private ManagedChannel createChannel() {
+    if (!customCredentialsType.equals("compute_engine_channel_creds")) {
+      throw new AssertionError(
+          "This test currently only supports "
+          + "--custom_credentials_type=compute_engine_channel_creds.");
+    }
+    ComputeEngineChannelBuilder builder = ComputeEngineChannelBuilder.forTarget(serverUri);
+    builder.keepAliveTime(3600, TimeUnit.SECONDS);
+    builder.keepAliveTimeout(20, TimeUnit.SECONDS);
+    return builder.build();
+  }
+
+  void setUp() {
+    channel = createChannel();
+    blockingStub = TestServiceGrpc.newBlockingStub(channel);
+  }
+
+  private void tearDown() {
+    try {
+      channel.shutdownNow();
+      channel.awaitTermination(1, TimeUnit.SECONDS);
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  private String streamToString(InputStream inputStream) throws Exception {
+    BufferedReader br = new BufferedReader(
+        new InputStreamReader(inputStream, UTF_8));
+    String out = "";
+    String line;
+    while ((line = br.readLine()) != null) {
+      out += line;
+    }
+    return out;
+  }
+
+  private void runShellCmd(String cmd) throws Exception {
+    logger.info("Run shell command: " + cmd);
+    Process process = Runtime.getRuntime().exec(new String[]{"bash", "-c", cmd});
+    int exitCode = process.waitFor();
```

Comment:
`waitFor()` should be put **after** draining the outputs. Otherwise the buffers may be full and that will block the child process. See:
https://wiki.sei.cmu.edu/confluence/display/java/FIO07-J.+Do+not+let+external+processes+block+on+IO+buffers

Reply:
Interesting, fixed